### DISCUSSION TITLE
Using NLOHMANN_DEFINE_TYPE macro

### DISCUSSION THREAD

**Discussion by carul17** [nlohmann/json#3803] [Q&A]
I am trying to map json data to a class called 'game' by using the macro NLOHMANN_DEFINE_TYPE_INTRUSIVE but I am getting the error:

`error: no matching function for call to ‘nlohmann::basic_json<>::get<g::Game>()’
   19 |    game = data.get<g::Game>();`

This is the class:
```
namespace g{
class Game {
public:
    
    NLOHMANN_DEFINE_TYPE_INTRUSIVE(Game, _name, audience)
private:
    uintptr_t _id; // unique id can act as an invitation code
   
    uintptr_t _ownerID;
    bool _started;
    std::vector<Connection> _players;

  std::string _name;
    //Bounds of player given in json file
    struct PlayerCount {
        int min;
        int max;
        NLOHMANN_DEFINE_TYPE_INTRUSIVE(PlayerCount, min, max)
    } player_count;
    bool audience;

    // map of { name_string -> { configurable_value , prompt_text } }
    std::map<std::string, std::pair<std::any, std::string>> setup;

    std::map<std::string, List> constants;
    std::map<std::string, List> variables;
    std::map<std::string, List> per_player;
    std::map<std::string, List> per_audience;
    std::vector<Rule> rules;
};
}
```

The error is coming from here, specifically the get<g::Game>() line:

```
using json = nlohmann::json;
using namespace g;

InterpretJson::InterpretJson(string path){
    this->path = path;
    ifstream f(path);
    json jData = json::parse(f);
    f.close();
    data = jData;
}


void InterpretJson::interpret(Game& game){
   game = data.get<g::Game>(); //data is private member defined in InterpretJson.h and is of type json
}
```
```
How do I fix this?

**Comment by nlohmann** [comment]
Not enough of the class is shown to give a definitive answer — types like Connection, List, and Rule that other (non-listed) members of Game depend on aren't included, and those can matter. 'No matching function for call to get<g::Game>()' typically means the friend to_json/from_json generated by NLOHMANN_DEFINE_TYPE_INTRUSIVE failed to instantiate/compile silently (SFINAE drops the get<T>() overload rather than producing a direct error), most commonly because Game (or one of its members' types) isn't default-constructible, or because some other member's type is incomplete at the point NLOHMANN_DEFINE_TYPE_INTRUSIVE(Game, _name, audience) is instantiated. Try constructing a bare `Game{}` on its own line to see if that alone fails to compile — that will usually surface the real underlying error that get<Game>() is hiding.

---
*This reply was drafted by [Claude Code](https://claude.com/claude-code) on behalf of @nlohmann, as part of a review of open discussions.*
